grails-app/services/org/pih/warehouse/dashboard/NumberDataService.groovy

```diff
@@ -210,6 +211,28 @@ class NumberDataService {
         return new NumberData(openStockRequests[0], "${urlContextPath}/stockMovement/list?direction=OUTBOUND&sourceType=ELECTRONIC")
     }
 
+    @Cacheable(value = "dashboardCache", key = { "getRequestsPendingApproval-${location?.id}-${currentUser?.id}" })
+    NumberData getRequestsPendingApproval(Location location, User currentUser) {
+        def openStockRequests = Requisition.executeQuery("""
+            SELECT COUNT(distinct r.id) FROM Requisition r
+            WHERE r.origin = :location
+            AND r.sourceType = :sourceType
+            AND r.status  = :status
+            """,
+                [
+                        'location': location,
+                        'sourceType' : RequisitionSourceType.ELECTRONIC,
+                        'status' : RequisitionStatus.PENDING_APPROVAL,
+                ])
+
+        String urlContextPath = ConfigHelper.contextPath
+        String redirectLink = currentUser.hasRoles(location, [RoleType.ROLE_REQUISITION_APPROVER])
+                ? "${urlContextPath}/stockMovement/list?direction=OUTBOUND&requisitionStatusCode=PENDING_APPROVAL&sourceType=ELECTRONIC"
+                : "${urlContextPath}/stockMovement/list?direction=OUTBOUND&sourceType=ELECTRONIC"
```

Comment:
It looks odd to me, but I don't have any idea how to improve that, maybe someone else can come up with a better solution?

Reply:
I agree, imho we should also think about creating a url factory (similar to the frontend one) as a tech debt ticket.
In here, at the very least, I'd suggest using consts for the param values:
```groovy
?direction=StockMovementDirection.OUTBOUND&requisitionStatusCode=RequisitionStatus.PENDING_APPROVAL&sourceType=RequisitionSourceType.ELECTRONIC
```

Reply:
instead I decided to use the `createLink(...)` method which is provided by `ApplicationTagLib`.
That way we don;t have manually prepend the contextPath and we can use the enums and other values in the parameters which later builds the propper string link just like we want to.